src/sentry/integrations/vsts/integration.py

```diff
@@ -81,6 +37,40 @@ def get_client(self):
             raise ValueError('Identity missing access token')
         return VstsApiClient(access_token)
 
+    def get_project_config(self):
+        client = self.get_client()
+        try:
+            projects = client.get_projects(self.model.metadata['domain_name'])
+        except ApiError:
+            # TODO(LB): This is an issue... what should happen if the client fails?
+            # should a message be shown to the user?
+            # Should we try refreshing the token? For VSTS that often clears up the problem
+            project_choices = []
```

Comment:
So in the case where `client.get_projects` fails, we will want it to do something. I did the bare minimum here, but let me know if we should tell the user something or try refreshing the token. I have a PR up for refreshing VSTS's token here: https://github.com/getsentry/sentry/pull/8725 Vsts appears to just throw a default unauthorized error even if it's just in need of a refresh token.

Reply:
Do we not know how long the token lasts? What I did in github over #8652 was refresh the token any time it expried.

Reply:
I've just been disabling the field if that happens. We can revisit that later.

Reply:
:) Ah! We do have that information. My bad; you're right! I didn't think about that.  I'll follow this up with another PR to refresh the data when it expires. Probably will follow after #8725.